src/main/java/org/mockbukkit/mockbukkit/entity/AbstractVillagerMock.java

```diff
@@ -0,0 +1,116 @@
+package org.mockbukkit.mockbukkit.entity;
+
+import org.bukkit.entity.AbstractVillager;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.Merchant;
+import org.bukkit.inventory.MerchantRecipe;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mockbukkit.mockbukkit.ServerMock;
+import org.mockbukkit.mockbukkit.inventory.MerchantInventoryMock;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Mock implementation of an {@link AbstractVillager}.
+ *
+ * @see AgeableMock
+ */
+public abstract class AbstractVillagerMock extends AgeableMock implements Merchant, AbstractVillager, InventoryHolder
+{
+	private final MerchantInventoryMock inventory = new MerchantInventoryMock(this, this);
+	private final List<MerchantRecipe> recipes = new ArrayList<>();
+
+	private @Nullable HumanEntity trader;
+
+	/**
+	 * Constructs a new {@link AbstractVillagerMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	protected AbstractVillagerMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+
+		updateTrades();
+	}
+
+	@Override
+	public @NotNull MerchantInventoryMock getInventory()
+	{
+		return this.inventory;
+	}
+
+	@Override
+	public void resetOffers()
+	{
+		this.recipes.clear();
+	}
+
+	@Override
+	public @NotNull List<MerchantRecipe> getRecipes()
+	{
+		return Collections.unmodifiableList(this.recipes);
+	}
+
+	@Override
+	public void setRecipes(@NotNull List<MerchantRecipe> recipes)
+	{
+		List<MerchantRecipe> clonedRecipes = recipes.stream()
+				.map(MerchantRecipe::new)
+				.toList();
+
+		this.recipes.clear();
+		this.recipes.addAll(clonedRecipes);
+	}
+
+	@Override
+	public @NotNull MerchantRecipe getRecipe(int i) throws IndexOutOfBoundsException
+	{
+
+		return new MerchantRecipe(this.recipes.get(i));
+	}
+
+	@Override
+	public void setRecipe(int i, @NotNull MerchantRecipe recipe) throws IndexOutOfBoundsException
+	{
+		this.recipes.set(i, new MerchantRecipe(recipe));
+	}
+
+	@Override
+	public int getRecipeCount()
+	{
+		return this.recipes.size();
+	}
+
+	@Override
+	public boolean isTrading()
+	{
+		return this.getTrader() != null;
```

Comment:
Same behavior as the default implementation. (You don't need this method at all)

Reply:
Where did you find the default implementation? I don't see the default implementation in the `org.bukkit.inventory.Merchant`:
![image](https://github.com/user-attachments/assets/5bb49594-55a0-4b5e-b0ef-a07ce4e073e8)

Reply:
Whoops. Might be a 1.21.4 thing.( I checked against 1.21.4)